qutebrowser/mainwindow/prompt.py

```diff
@@ -642,8 +643,16 @@ def _init_fileview(self):
         self._file_model.directoryLoaded.connect(
             lambda: self._file_model.sort(0))
 
+    def _show_error(self, msg):
+        log.prompt.error(msg)
+        QToolTip.showText(self._lineedit.mapToGlobal(QPoint(0, 0)), msg)
+
     def accept(self, value=None):
         text = value if value is not None else self._lineedit.text()
+        text = downloads.transform_path(text)
+        if text is None:
+            self._show_error("Invalid filename")
+            return False
```

Comment:
I'm not sure if this should be here. Right now, we only use `DownloadFilenamePrompt`, but we don't know yet for what we'd use `FilenamePrompt` in the future, and if this would be appropriate.

Alternatively, we could keep it, but then `transform_path` should probably go to `utils`.